server/forward.go

```diff
@@ -0,0 +1,13 @@
+package server
+
+import (
+	"github.com/ipfs/go-bitswap/server/internal/decision"
+)
+
+type (
+	Receipt                = decision.Receipt
+	PeerBlockRequestFilter = decision.PeerBlockRequestFilter
+	TaskComparator         = decision.TaskComparator
+	ScoreLedger            = decision.ScoreLedger
+	ScorePeerFunc          = decision.ScorePeerFunc
+)
```

Comment:
moving this from /decision to /server will break consumers. I'm okay with breaking, but if we break then we might as well do the options right too. Or don't break them and don't break this.

Reply:
@guseggert thoses are just interfaces, I'm not sure what you want to change here ?

Reply:
The interfaces in `/decision` are exported and consumed by other modules (such as [in kubo](https://github.com/ipfs/kubo/blob/master/core/commands/bitswap.go#L12)), moving them breaks such consumers. If we're going to break backwards compat then we might as well do the options right too (use real types instead of `interface{}`). Or just keep backwards compat here (which is what I would do).

Reply:
@guseggert

1. I don't want to expose the real types because they have a bunch of internal stuff that consumers shouldn't mess around with.
2. Thoses interfaces are all (except Receipt) callbacks the consumers can use to customize their decision manager. There is no "real types", thoses things are meant to be implemented by consumers.